pkg/planner/cardinality/row_count_column.go

```diff
@@ -182,6 +182,10 @@ func equalRowCountOnColumn(sctx planctx.PlanContext, c *statistics.Column, val t
 		if modifyCount == 0 {
 			return 0, nil
 		}
+		newRows := float64(realtimeRowCount) - c.TotalRowCount()
+		if newRows < 0 {
```

Comment:
May I ask what the purpose of this change is?

Reply:
Also, why do we need to use the total row count instead of the null count here?

Reply:
The change is simply to avoid a negative number. The reason is that if there have been modifications - then the original stats are now unreliable. If realtimeRowCount has increased - then it is possible that the value we are searching for is in those added rows. If decreased - it could be that they are ONLY deletes - which means that the value we are searching for isn't in the data - but it also could be a combination of inserts/updates/deletes - we simply do not know.

I use TotalRowCount rather than notNullCount because - if there are NULLs - then notNullCount may always be lower than realtimeRowCount. Because realtimeRowCount counts all rows (I believe). So using notNullCount would incorrectly have us counting those "NULL rows" here.